pkgs/applications/editors/emacs/make-emacs.nix

```
@@ -88,6 +89,8 @@
   withTreeSitter ? lib.versionAtLeast version "29",
   withWebP ? lib.versionAtLeast version "29",
   withX ? !(stdenv.hostPlatform.isDarwin || noGui || withPgtk),
+  withCairo ? withX && !withXft,
+  withXft ? false,
```

```suggestion
  withCairo ? true,
  withXft ? withX && !withCairo,
```
This way we can ensure some "maximality" by default

---

At first, I thought that Cairo was enabled by default, and manually enabling Xft would disable Cairo, so I wrote it this way. But since that's the case, I will modify it later.

---

But if that's the case, when someone manually enables Xft without specifying withCairo = false;, the checks in configure.ac will always prioritize using Cairo.

---

So should we use withXft = true or withCairo = false?

---

Well, given that Cairo overrides Xft, your idea is better. 
Let's forget this suggestion.